app/screens/post_options/bindings/index.ts

```diff
@@ -24,9 +25,22 @@ type OwnProps = {
     post: Post;
 }
 
+const getPostMenuBindings = makeAppBindingsSelector(AppBindingLocations.POST_MENU_ITEM);
+const getThreadPostMenuBindings = makeRHSAppBindingSelector(AppBindingLocations.POST_MENU_ITEM);
+
 function mapStateToProps(state: GlobalState, props: OwnProps) {
     const apps = appsEnabled(state);
-    const bindings = apps ? getAppsBindings(state, AppBindingLocations.POST_MENU_ITEM) : [];
+    let bindings: AppBinding[] | null = [];
+    if (apps) {
+        if (props.post.channel_id === getCurrentChannelId(state)) {
+            bindings = getPostMenuBindings(state);
+        } else if (props.post.channel_id === getThreadAppsBindingsChannelId(state)) {
+            bindings = getThreadPostMenuBindings(state);
+        } else {
+            bindings = null;
+        }
+    }
```

Comment:
Should we set the initial value to `null`, as we are setting at the end of this if block? Making a new array for the initial every time this function runs will cause a re-render if apps aren't enabled.

Reply:
The semantics of null here would not be valid, since if we pass null, the component will try to fetch its own bindings. What I can do is declare the empty bindings as a constant, and use that.